### ISSUE TITLE
Can't reproduce reported results

### ISSUE THREAD

**Issue by ruoshiliu** [ferrine/hyrnn#17]
Hi, I ran the exact argument you provided in the README.md file
`python run.py --data_dir=./data --num_epochs=30 --log_dir=./logs --batch_size=1024 --num_layers=2 --cell_type=hyp_gru`
which yields a loss of **0.678** and precision of **~56%**
I also ran
`python run.py --data_dir=./data --num_epochs=30 --log_dir=./hyp_gru_hyp_decision --batch_size=1024 --num_layers=2 --cell_type=hyp_gru --decision_type hyp`
which yields a loss of **0.693** and precision of **~50%**
Do you know any potential reason for that?

**Comment by ferrine** [comment]
hmm, that is strange. We had updates in Geoopt and implementation of Poincare model is different now, this might affect the results. Did you check the old version of geoopt (before Stereographic is merged)?

**Comment by ruoshiliu** [comment]
Yes, I installed version 0.1.2 of geeopt which is before the Stereographic update.

**Comment by ferrine** [comment]
That's a problem smth really seems to be broken. How does the training curve look like?

**Comment by ruoshiliu** [comment]
Please check the following tensorboard results, with orange being train curves and blue being valid curves. I obtained the following results with `python run.py --data_dir=./data --num_epochs=30 --log_dir=./logs --batch_size=1024 --num_layers=2 --cell_type=hyp_gru`
![Screen Shot 2020-07-28 at 13 45 30](https://user-images.githubusercontent.com/25470644/88708104-c1bb1c80-d0d8-11ea-9381-e1a9c4084f37.png)
![Screen Shot 2020-07-28 at 13 45 39](https://user-images.githubusercontent.com/25470644/88708110-c41d7680-d0d8-11ea-836b-273daabe425f.png)
![Screen Shot 2020-07-28 at 13 46 01](https://user-images.githubusercontent.com/25470644/88708117-c7b0fd80-d0d8-11ea-9d25-1a8dce03c53a.png)

**Comment by SkyFishMoon** [comment]
It occurs also in the new version. But I found that if decode using method in another paper named Hyperbolic GCN, the precision will arise a lot, but still can't get the result in original paper

**Comment by ferrine** [comment]
From the curves I can't say the model has converged, did you try training more?

**Comment by ferrine** [comment]
Other suggestion is to increase learning rate a bit.

**Comment by SkyFishMoon** [comment]
This is the validation 
![image](https://user-images.githubusercontent.com/50042586/91652431-e20c3b80-eac9-11ea-8b3f-f0492ef1cdf9.png)
and train result
![image](https://user-images.githubusercontent.com/50042586/91652451-154eca80-eaca-11ea-84e9-e71d9ed1beb0.png)
a little rise in precision, but I didn't use the dist2plane as the last layer. I just use a linear + logmap0 + logsoftmax + nllloss as the last layer.
Another problem is I find that some parameters in your model are general torch.Parameter rather than geoopt.ManifoldParameter. Only the bias parameters are geoopt.ManifoldParameter. Does this matter?
